src/vs/code/node/cli.ts

```diff
@@ -319,23 +324,61 @@ export async function main(argv: string[]): Promise<any> {
 			options['stdio'] = 'ignore';
 		}
 
-		const child = spawn(process.execPath, argv.slice(2), options);
+		let child: ChildProcess;
```

Comment:
I suggest to put some `//` comments for the if/else here:
* Windows, Linux: electron executable is `spawn`ed directly
* macOS: using the `open` command (explain why)